src/shims/unix/foreign_items.rs

```diff
@@ -894,7 +894,8 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
                 let _attr_place =
                     this.deref_pointer_as(attr_place, this.libc_ty_layout("pthread_attr_t"))?;
                 let addr_place = this.deref_pointer(addr_place)?;
-                let size_place = this.deref_pointer(size_place)?;
+                let size_place =
+                    this.deref_pointer_as(size_place, this.libc_ty_layout("size_t"))?;
```

Comment:
Below we pretty much hard-code this to be ptr-sized, so you might as well use `this.machine.layouts.usize`.

And then please adjust `addr_place`  just above as well so that we are done with this particular shim. :)